### DISCUSSION TITLE
Problem installing nextflow on wsl: Error: Could not find or load main class NextflowLoader

### DISCUSSION THREAD

**Discussion by Iffy85** [nextflow-io/nextflow#2728] [Q&A]
Am having issues installing nextflow on wsl in windows 10 Pro
After successfully executing command **wget -qO- https://get.nextflow.io | bash** and **chmod +x nextflow**
when execute nextflow following error arises:
**Error: Could not find or load main class NextflowLoader
Unable to initialize nextflow environment**

I have also tried installing complete assest using **wget -qO- ASSET-https://github.com/nextflow-io/nextflow/releases/download/v22.02.1-edge/nextflow-22.02.1-edge-all**
Still the error remains the same

**Comment by bentsherman** [comment]
1. Which linux distro are you running?
2. Which version of java do you have?
3. Try running nextflow with extra logging:
  ```bash
  nextflow -Dcapsule.log=verbose info 
  ```
